---
title: "Supplemental Traditional Analyses"
author: 
- \* Emily Smenderovac (corresponding author), Great Lakes Forestry Centre, Natural Resources Canada, emily.smenderovac@nrcan-rncan.gc.ca
- Jesse Hoage,  Laurentian University, Natural Resources Canada, Jesse.hoage@gmail.com
- Teresita M. Porter, Centre for Biodiversity Genomics and Department of Integrative Biology, University of Guelph terrimporter@gmail.com
- Caroline Emilson, Great Lakes Forestry Centre, Natural Resources Canada, caroline.emilson@nrcan-rncan.gc.ca
- Rob Fleming, Great Lakes Forestry Centre, Natural Resources Canada, rob.fleming@nrcan-rncan.gc.ca
- Nathan Basiliko, Laurentian University, nbasiliko@laurentian.ca
- Merhdad Hajibabei, Centre for Biodiversity Genomics and Department of Integrative Biology, University of Guelph mhajibab@uoguelph.ca
- Dave Morris, Centre for Northern Forest Ecosystem Research, Ontario Ministry of Natural Resources and Forestry, Thunder Bay CA P7E 2V6, Dave.M.Morris@ontario.ca
- Lisa Venier, Great Lakes Forestry Centre, Natural Resources Canada, lisa.venier@nrcan-rncan.gc.ca
date: "`r Sys.Date()`"
bibliography: bibliography.json
csl: forest-ecology-and-management.csl
output: 
      bookdown::html_document2:
            toc: true
            number_sections: true
            keep_md: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)

memory.limit(size=30000) ## increase R memory limit

## grab essential libraries
library(tidyverse)
library(ALDEx2)
library(vegan)
library(pheatmap)
library(compositions)
library(readr)
library(ggpubr)
source("src/HelperFunctions_CommunityData.R")
## Load data

for(d in list.files("data", full.names = T)[grepl("RData$", list.files("data", full.names = T))]){
      load(d)
}
rm(d)

metadat$month <- factor(metadat$month, levels = c("June", "August", "October"))
metadat$harvest <- factor(metadat$harvest, levels = c("Full Tree", "Unharvested", "Stem Only", "Stumped", "Bladed"))
metadat$topo[metadat$harvest == "Bladed"] <- "Bladed"


### Prepare taxonomy for Functional assignment information for the different zOTUS
## ITS
FunGuild_data <- ITS[,c("ITS_GlobalESV", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")]

colnames(FunGuild_data)[1] <- "OTU ID"

write_delim(FunGuild_data, file("FUNGuild/FunGuild.taxa.txt", encoding="UTF-8"), delim="\t")
 rm(FunGuild_data)

 ## Bacterial
 FaproTax_data <- Bac16S
 colnames(FaproTax_data)[1] <- "OTU ID"

FaproTax_data$Genus <- ifelse(FaproTax_data$gBP >=0.5, FaproTax_data$Genus, "unidentified") 
 ## No Species in bac df

FaproTax_data$taxonomy <- paste(paste("d", FaproTax_data$Domain, sep="__"), 
                                paste("p", FaproTax_data$Phylum, sep="__"),
                                paste("c", FaproTax_data$Class, sep="__"),
                                paste("o", FaproTax_data$Order, sep="__"),
                                paste("f", FaproTax_data$Family, sep="__"),
                                paste("g", FaproTax_data$Genus, sep="__"),
                                sep=";")

FaproTax_data<- FaproTax_data[,c("taxonomy", "OTU ID")]

write_delim(FaproTax_data, file("FAPROTAX_1.2.4/FaproTax_data.tsv", encoding="UTF-8"), delim="\t")
 

## Arthropods
BETSI_search <- str_c(gsub('^undef_', '', c(unique(F230$Phylum), unique(F230$Class), unique(F230$Family), unique(F230$Genus[F230$gBP >= 0.3]), gsub("_", " ", unique(F230$Species[F230$sBP >= 0.7])))),
                      collapse=";")


BETSI_search1 <- str_c(c(unique(F230$Class), unique(F230$Family), unique(F230$Genus[F230$gBP >= 0.3])),
                      collapse=";")

BETSI_species <-  paste(c(gsub("_", " ", unique(F230$Species[F230$sBP >= 0.7]))), ";", sep="")
write(c("Taxon_name", BETSI_species), "BETSI/BETSI_Species_check.csv")

BETSI_species <- readLines("BETSI/transform.csv")[3:81]

species_corrected <- data.frame()

for(sp in BETSI_species){
  species_corrected <-  rbind(species_corrected, 
                              data.frame(original.species = str_split(sp, ";")[[1]][1], Corrected = str_split(sp, ";")[[1]][2]))
}

not_in_BETSI <- paste(sum(species_corrected$Corrected == "")/nrow(species_corrected)*100, "%")

species_corrected <- species_corrected[!species_corrected$Corrected == "", ]

BETSI_species_search <-  str_c(gsub("\t","", species_corrected$Corrected), collapse = ";")

##Conducted search & correction on https://portail.betsi.cnrs.fr/ for diet level 2 and microhabitat level 2

rm(species_corrected, not_in_BETSI, sp, BETSI_species)

## compile metabarcode datasets
metabarcode_sets <- list(Bac16S = Bac16S, ITS = ITS, F230 = F230, Euk18S = Euk18S)
rm(Bac16S, F230, Euk18S, ITS)


ds_list <- list()
count = 1
## Set up metabarcoding datasets for different hypothesis tests
for(ds in c("Bac16S", "ITS", "F230", "Euk18S")){
   for(set in 1:3){
      if(set == 1){toexclude = metadat$sample[!metadat$month == "August"]}
      if(set == 2){toexclude = metadat$sample[!(metadat$harvest %in% c("Unharvested", "Full Tree") & is.na(metadat$topo))]}
      if(set == 3){toexclude = metadat$sample[is.na(metadat$topo)]}
      
      dat = metabarcode_sets[[ds]]
      
      colnames(dat)[1] <- "ESV"
      
      ds_subset <- dat %>% 
         filter(!SampleName %in% toexclude) %>%
         group_by(ESV) %>%
         filter(!sum(ESVsize) == 0) %>% 
         pivot_wider(names_from = SampleName, values_from = ESVsize, values_fill = 0) %>% 
         data.frame()
      
      ds_list[[count]] <- ds_subset
      names(ds_list)[count] <- paste0(ds, "_", set)
      count=count+1
      
      
      ### Add functional & Genus grouping levels as well --> maybe bring that in at a later point if anything found with ESV
   }   
}

rm(metabarcode_sets, ds_subset, dat, ds, count, set, toexclude)

## Set a colorblind-safe palette
cust.cols <- c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255")

num_p_tests <- 0
figcount <- 1
```

```{asis, eval=FALSE}
## Conducted on Powershell on Windows
## FUNGuild fungi functional assignment
cd FUNGuild

python FUNGuild.py guild -taxa FunGuild.taxa.txt

## Conducted on Powershell on Windows
cd ..\FAPROTAX_1.2.4

## Replaced is not statements with "!="

python collapse_table_v2.py -i FaproTax_data.tsv -g FAPROTAX.txt -f -o functional_otu_table.tsv -r report.txt --column_names_are_in first_data_line  --keep_header_comments --non_numeric consolidate -v --row_names_are_in_column "taxonomy" --omit_columns 0 --normalize_collapsed none --group_leftovers_as 'other'

## Matching was poor- as no id's to species - only 6087, 21452 leftovers

```


```{r taxa-inf}
## Fungi
FunGuild_data <- read.delim("FUNGuild/FunGuild.taxa.guilds.txt")

## Bacteria
FaproTax_data <- read.delim("FAPROTAX_1.2.4/functional_otu_table.tsv", skip = 2)

## Arthropods
## Pull in the Class, Order, Genus search results for diet (includes species within the genus)
unzip("BETSI/BETSI_0_cog_diet.zip", exdir="BETSI")

BETSI_db_res <- read.delim("BETSI/BETSI_A.csv")
BETSI_db_mat <- read.delim("BETSI/BETSI_B.csv")

BETSI_db_mat <- BETSI_db_mat[, 1:(ncol(BETSI_db_mat)-2)]

col_repl <- data.frame(orig=colnames(BETSI_db_mat), row.2 = t(BETSI_db_mat[1,])) %>%
  mutate(grp = cumsum(!grepl("X", orig))) %>%
  group_by(grp) %>%
  mutate(first.term=orig[!grepl('X', orig)])

colnames(BETSI_db_mat) <- c("Taxon", paste(col_repl$first.term, col_repl$X1, sep=".")[2:ncol(BETSI_db_mat)])

BETSI_db_mat_diet <- BETSI_db_mat[2:nrow(BETSI_db_mat),] %>% pivot_longer(-Taxon, values_to = "presence", names_to = "Function")

## Pull in the Class, Order, Genus search results for diet (includes species within the genus)
unzip("BETSI/BETSI_0_cog_microhabitat.zip", exdir="BETSI")

BETSI_db_res <- read.delim("BETSI/BETSI_A.csv")
BETSI_db_mat <- read.delim("BETSI/BETSI_B.csv")

BETSI_db_mat <- BETSI_db_mat[, 1:(ncol(BETSI_db_mat)-2)]

col_repl <- data.frame(orig=colnames(BETSI_db_mat), row.2 = t(BETSI_db_mat[1,])) %>%
  mutate(grp = cumsum(!grepl("X", orig))) %>%
  group_by(grp) %>%
  mutate(first.term=orig[!grepl('X', orig)])

colnames(BETSI_db_mat) <- c("Taxon", paste(col_repl$first.term, col_repl$X1, sep=".")[2:ncol(BETSI_db_mat)])

BETSI_db_mat_microhabitat <- BETSI_db_mat[2:nrow(BETSI_db_mat),] %>% pivot_longer(-Taxon, values_to = "presence", names_to = "Function")

## Combine trait Matrixes

BETSI_db_mat_combined <- rbind(BETSI_db_mat_diet, BETSI_db_mat_microhabitat)

##

BETSI_db_mat_combined <- BETSI_db_mat_combined %>%
  filter(!presence == 0) %>%
  group_by(Taxon) %>%
  summarize(Function = str_c(unique(Function), collapse = "-")) 


rm(BETSI_db_mat_microhabitat, BETSI_db_mat, BETSI_db_mat_diet, BETSI_db_res, col_repl)

```


```{r harvest-ortho, include=FALSE}
  ### Set defined contrasts
  ## 1: Harvested vs Uncut control - SO FT S vs Control
  ## 2: forest floor rremoval vs retention -  Bladed vs SO. FT. S.
  ## 3: Stump/roots retained vs. removed - S vs SO, FT
  ## 4: only coarse wood vs Coarse & fine woody debris  (FT vs SO)
      harvest.contrast.matrix = data.frame(TR = c(`Full Tree` = 0.33, Unharvested = -1, `Stem Only` = 0.33, `Stumped` = 0.33, `Bladed` = 0), 
                                   FFR = c(`Full Tree` = -0.33, Unharvested = 0, `Stem Only` = -0.33, `Stumped` = -0.33, `Bladed` = 1), 
                                   BGBR = c(`Full Tree` = -0.5, Unharvested = 0, `Stem Only` = -0.5, `Stumped` = 1, `Bladed` = 0),
                                   BRt = c(`Full Tree` = 1, Unharvested = 0, `Stem Only` = -1, `Stumped` = 0, `Bladed` = 0))

harvest.ortho.table <- data.frame(c(Code = "TR", Contrast = "Stem-Only, Full tree and Stumped vs Uncut control", Tests = "Effect of tree removal"),
                                  c(Code = "FFR", Contrast = "Bladed vs Stem-Only, Full tree and Stumped", Tests = "Effect of forest floor removal"),
                                  c(Code = "BGBR", Contrast = "Stumped vs Stem-Only and Full tree", Tests = "Effect of below-ground woody biomass removal"),
                                  c(Code = "BRt", Contrast = "Full tree vs Stem-Only", Tests = "Effect of fine and coarse woody debris retention vs only coarse wood")) %>% t()

rownames(harvest.ortho.table) <- NULL

knitr::kable(harvest.ortho.table, caption= "Orthagonal contrasts used to test effects of intensified harvesting.")
```


```{r topo-ortho, include=FALSE}
  ### Set defined contrasts
 
      topo.contrast.matrix = data.frame(T.B = c(`Bladed` = -1, Flat = 0, `Pile` = 0, `Trench` = 1), 
                                        T.PF = c(`Bladed` = 0, Flat = -0.5, `Pile` = -0.5, `Trench` = 1),
                                        P.F = c(`Bladed` = 0, Flat = -1, `Pile` = 1, `Trench` = 0))

topo.ortho.table <- data.frame(c(Code = "T.B", Contrast = "Trench vs Bladed", Tests = "Are effects of forest floor removal buffered in trenches"),
                                  c(Code = "T.PF", Contrast = "Trench vs Piles and Flats", Tests = "Effect of forest floor removal from trenches"),
                                  c(Code = "P.F", Contrast = "Pile vs Flats", Tests = "Effect of mixing of soil layers in piles")) %>% t()

rownames(topo.ortho.table) <- NULL

knitr::kable(topo.ortho.table, caption= "Orthagonal contrasts used to test differences between topographic features.")
```



# Results
```{r prep_CODA, cache=TRUE, include = FALSE}

source("src/HelperFunctions_CommunityData.R")

coda.clrs.res <- list()
count= 1
for(ds_n in 1:length(ds_list)){
        ds_name <- names(ds_list)[ds_n]
        
          Orddf <- ds_list[ds_name][[1]]
          Orddf <- Orddf[, colnames(Orddf) %in% metadat$sample]
  
  if(grepl("Bac", ds_name)){
        Orddf <- t(apply(Orddf, 2, function(x){x/sum(x)})) ## Relative abundance
  }else{
         Orddf <- t(apply(Orddf, 2, function(x){as.numeric(x>0)})) ## presence/absence
  }
                coda.clrs.res[[count]] <- list(Orddf)
                names(coda.clrs.res)[count] <- paste(ds_name, sep="_")
                count=count+1
          
                
        
}

rm(count, ds_name, Ord.clr, Orddf4aldex, subset_aldex, Orddf, ds_n)

```


```{r coda-pca-var, include=TRUE, fig.cap= "Community variance of relative abundance Bray-Curtis (16S) or presence-absence Jaccard (ITS, F230, 18S) distance explained in first two PC axes for each metabarcoding dataset.", echo=FALSE}

clr.pca.plotting <- data.frame()

for(ds_n in 1:length(coda.clrs.res)){
        ds_name <- names(coda.clrs.res)[ds_n]
        
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ## SVD PCA of clr values
        pca <- prcomp(d.clr.abund)
        
        clr.pca.plotting <- rbind(clr.pca.plotting, data.frame(dataset = ds_name, variance=pca$sdev^2/sum(pca$sdev^2)))
}

pca.plotting <- clr.pca.plotting %>%
  mutate(temp = 1) %>%
  group_by(dataset) %>%
  mutate(PCA = cumsum(temp), max.var = max(variance)) %>% 
  filter(PCA %in% c(1,2)) %>%
  summarize(variance=sum(variance)) %>%
  arrange(str_extract(dataset, ".$"), desc(variance)) %>%
  mutate(dataset = factor(as.character(dataset), levels = unique(as.character(dataset))))
  ggplot(pca.plotting, aes(variance, dataset)) + geom_bar(stat= "identity") + xlab("Variance Explained by PC1 and PC2") + theme_minimal()
  
rm(clr.pca.plotting, pca, char_data, ds_n, ds_name, d.clr.abund, pca.plotting)
```

More community variance was explained in traditional approaches for bacterial communities compared to compositional approaches, but traditional approaches had less explained variance for eukaryote (F230, ITS, 18S) communities. 


## Intensified Harvesting


### Differences in diversity

```{r CODA-PCA-1, include=FALSE}
clr.pca.plots <- list()

plotcount <- 1

for(ds_name in names(coda.clrs.res[grep("_1", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        if(grepl("Genus|functional", ds_name)){next}
      
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##PCA of the clr values
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ]# %>%
          # mutate(harvest = factor(as.character(harvest), levels = c("Unharvested", "Stem Only", "Full Tree", "Stumped", "Bladed")))
        
           pca <- prcomp(d.clr.abund)
        
        pca.var.explained <- pca$sdev^2/mvar(d.clr.abund)
        
        ## Prepare plotting variables
        xlab.val <- paste("PCA1", round(pca.var.explained[1], 3))
        ylab.val <- paste("PCA2", round(pca.var.explained[2], 3))
        
        ## extract the same way base biplot for princomp works
        plottingdat <- as.data.frame(pca$x/ncol(pca$x)) %>%
          dplyr::select(PC1, PC2) %>%
          rownames_to_column("sample") %>% 
          left_join(char_data, by="sample") %>%
          dplyr::select(-sample)
        
         ## Find Hulls of data
      hulls = Ordination_hulls(plottingdat, c("harvest")) 
      
       clr.pca.plots[[plotcount]] <-ggplot(plottingdat, aes(PC1, PC2, group=harvest))+
   geom_point(data=plottingdat, aes(color=harvest, shape = harvest), alpha=0.5, size = 2) + 
   geom_polygon(data=hulls, aes(group=hullgrp, fill = harvest, alpha = harvest, color=harvest)) +
   scale_color_manual(values= cust.cols[c(2, 3, 1, 6, 8)], breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   scale_fill_manual(values= cust.cols[c(2, 3, 1, 6, 8)], breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   scale_shape_manual(values= c(2, 3, 1, 5, 6), breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   scale_alpha_manual(values= c(0.2, 0.2, 0.2, 0.2, 0.2), breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   xlab(xlab.val)+
   ylab(ylab.val)+
   theme_minimal()+
     ggtitle(ds_name) 
   
       #print(clr.pca.plots[[plotcount]])
       
names(clr.pca.plots)[plotcount] <- ds_name
        plotcount <- plotcount +1
}


   
rm(plotcount, ds_name, char_data, d.clr.abund, hulls, pca, plottingdat, pca.var.explained, xlab.val, ylab.val)
```

### Community PCA plots {.tabset}

#### Bacterial (16S) {.unnumbered}

```{r PCA1}
clr.pca.plots[[1]]
```

#### Fungal (ITS) {.unnumbered}

```{r PCA2}
clr.pca.plots[[2]]
```

#### Arthropods (F230){.unnumbered}

```{r PCA3}
clr.pca.plots[[3]]
```

#### Eukaryotes (18S){.unnumbered}

```{r PCA4}
clr.pca.plots[[4]]

rm(clr.pca.plots)
```






```{r CODA-PERMANOVA, include=FALSE}
clr.PERM.res <- list()
clr.adon.res <- data.frame()
clr.betadisper.res <- data.frame()
count <- 1

for(ds_name in names(coda.clrs.res[grep("_1", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        #if(grepl("Genus|functional", ds_name)){next}
      
        if(grepl("Bac", ds_name)){
        distmetric = "bray" ## Relative abundance
  }else{
         distmetric = "jaccard" ## presence/absence
  }
               
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##char data
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ]
        
        ##OVERALL PERMANOVA
        adon.res.tmp <- adonis(d.clr.abund~harvest, data=char_data, method = distmetric)$aov.tab %>%
           rownames_to_column("Parameter") %>%
           mutate(contrast = "overall",
                  dataset = ds_name) %>% data.frame()
        
        ##OVERALL BETADISPER
        betadisper.res.tmp <- anova(betadisper(vegdist(d.clr.abund, method = distmetric), char_data$harvest)) %>%
           rownames_to_column("Parameter") %>%
           mutate(contrast = "overall",
                  dataset = ds_name) %>% data.frame()
        
        ##PAIRWISE PERMANOVA
        adon.res.tmp <- ortho_adonis(d.clr.abund, harvest.contrast.matrix, c("harvest"), char_data, method=distmetric) %>%
           mutate(dataset = ds_name, Parameter = "") %>% rbind(adon.res.tmp)
        
        ##PAIRWISE BETADISPER
        
          betadisper.res.tmp <- ortho_betadisper(d.clr.abund, harvest.contrast.matrix, c("harvest"), char_data, method=distmetric) %>%
           mutate(dataset = ds_name, Parameter = "") %>% rbind(betadisper.res.tmp)
       
          
         clr.adon.res <- rbind(clr.adon.res, adon.res.tmp)
         clr.betadisper.res <- rbind(clr.betadisper.res, betadisper.res.tmp)
         
         
         clr.PERM.res[[count]] <- list(adonis = adonis(d.clr.abund~harvest, data=char_data,  method = distmetric), betadisper = betadisper(vegdist(d.clr.abund,  method = distmetric), char_data$harvest))
         names(clr.PERM.res)[count] <- ds_name
         count = count + 1
}

num_p_tests <- num_p_tests + length(unique(clr.adon.res$dataset)) + length(unique(clr.adon.res$dataset))
   
rm(count, ds_name, adon.res.tmp, betadisper.res.tmp, char_data, d.clr.abund)
```

#### PERMANOVA results {.unnumbered}

```{r CODA-adonis-plot-1, fig.cap = "Results of community structure PERMANOVA tests performed with the orthogonal contrasts described in Table \\@ref(tab:harvest-ortho) (TR = harvesting vs uncut, FFR = forest floor removal, BGBR = below-ground biomass removal, BRt = biomass retention). The variance explained by each comparison (R^2^) is shown on the x axis, and the probability of the effect is shown on the y axis (Pr (>F)). A black line is used to display p = 0.05.", fig.width = 8}
adon.plotting <- clr.adon.res[!clr.adon.res$Parameter %in% c("Residuals", "Total", "harvest") , ] %>%
      mutate(contrast = factor(as.character(contrast), levels = c("overall", "TR", "BRt", "BGBR", "FFR")))
adon.plotting$dataset <- gsub("_.$", "", adon.plotting$dataset)

ggplot(adon.plotting, aes(R2, Pr..F., color=contrast, fill = contrast, shape = contrast)) +
   geom_point(size = 3) + 
   geom_hline(yintercept = 0.05, color="black") +
   facet_wrap(~dataset, scales = "free") +  
   scale_color_manual(values= c("black", cust.cols[c(2, 3, 6, 8)]))+
   scale_fill_manual(values= c("black", cust.cols[c(2, 3, 6, 8)]))+
   scale_shape_manual(values= c(2, 3, 5, 6, 8))+
   theme_minimal() +
   xlab(expression("Effect Size (R"^2*")")) + ylab("Pr (>F)")
   
rm(clr.adon.res, adon.plotting)
```


#### Betadispersion {.unnumbered} 

```{r CODA-bd-plot-1, fig.cap="Results of betadispersion tests comparing community structures of the different harvesting treatments. The betadispersion of each treatment is shown on the y axis, and the treatment is shown along the x axis. Significant contrast tests as outlined in Table \\@ref(tab:harvest-ortho) (TR = harvested vs uncut, FFR = forest floor removal, BGBR = below-ground biomass removal, BRt = biomass retention) are indicated along the top of the figure with lines and symbols designating test significance (* = p < 0.05, ** = p < 0.01, *** = p < 0.001).", fig.width= 8, fig.height = 4}

plot.betadisper <- data.frame()

for(ds_name in names(clr.PERM.res)){
   temp <- clr.PERM.res[[ds_name]]$betadisper$distances
   
   ## pull distances for plotting
   temp <- data.frame(temp, sample = names(temp)) %>%
     left_join(metadat, by = "sample") %>%
     mutate(dataset=ds_name) %>%
     rename(distance = temp) %>%
     mutate(harvest = factor(as.character(harvest), levels = c("Unharvested", "Stem Only", "Full Tree", "Stumped", "Bladed")))
   
   plot.betadisper <- rbind(plot.betadisper, temp)
     
     
}

## Get min distances for each set
mins  <- plot.betadisper %>%
  group_by(dataset) %>%
  summarize(distance = min(distance))

## Get max distances for each set
maxs  <- plot.betadisper %>%
  group_by(dataset) %>%
  summarize(distance = max(distance))

 ### Pull test result
     temp.sig <- clr.betadisper.res %>%
      filter(contrast != "overall") %>%
     dplyr::select(-Parameter) %>%
           mutate(firstseg.start = ifelse(contrast == "TR", 
                                   0.75, 
                                   ifelse(contrast == "FFR", 1.75, 
                                          ifelse(contrast == "BGBR", 1.75, 1.75))), 
                  firstseg.end = ifelse(contrast == "TR", 
                                   1.25, 
                                   ifelse(contrast == "FFR", 4.25, 
                                          ifelse(contrast == "BGBR", 3.25, 2.25))), 
                  secondseg.start = ifelse(contrast == "TR", 
                                   1.75, 
                                   ifelse(contrast == "FFR", 4.75, 
                                          ifelse(contrast == "BGBR", 3.75, 2.75))), 
                  secondseg.end = ifelse(contrast == "TR", 
                                   4.25, 
                                   ifelse(contrast == "FFR", 5.25, 
                                          ifelse(contrast == "BGBR", 4.25, 3.25)))) %>%
       group_by(dataset) %>%
     mutate(sig.level = ifelse(Pr..F. < 0.001, paste("***", contrast), ifelse(Pr..F. < 0.01, paste("**", contrast), ifelse(Pr..F. < 0.05, paste("*", contrast), contrast))), 
            text.col = ifelse(grepl("\\*", sig.level), "significant at p-value < 0.05", "non-significant")) %>%
       left_join(maxs, by = "dataset") %>%
           mutate(dataset = gsub("_[[:digit:]]", "", dataset), 
                  distance = ifelse(contrast == "TR", 
                                   distance * 1.4, 
                                   ifelse(contrast == "FFR", distance* 1.3, 
                                          ifelse(contrast == "BGBR", distance*1.2, distance*1.1))))

## plot as boxplots with significance
     ggplot(plot.betadisper %>% mutate(dataset = gsub("_[[:digit:]]", "", dataset)), aes(harvest, distance, fill = harvest))+
     geom_boxplot(alpha = 0.5)+
       geom_text(data= temp.sig, inherit.aes = F, aes(5.75, distance*0.9, label = sig.level, color = text.col))+
           geom_segment(data = temp.sig, inherit.aes = F, aes(y = distance*0.9, yend = distance*0.9, x = firstseg.start, xend = firstseg.end, color = text.col)) +
           geom_segment(data = temp.sig, inherit.aes = F, aes(y = distance*0.9, yend = distance*0.9, x = secondseg.start, xend = secondseg.end, color = text.col)) +
       theme_minimal()+
       scale_fill_manual(values= c(cust.cols[c(2, 3, 1, 6, 8)]), breaks = levels(temp$harvest)) +
       scale_color_manual(name = "significance", values= c("black", "red")) +
       facet_wrap(~dataset, scales = "free") + coord_cartesian(xlim = c(0, 5.8))
     
     rm(plot.betadisper, temp.sig, mins, clr.betadisper.res, clr.PERM.res, ds_name)
```




## {-}


While there was a single stem only outlier point in bacterial community structure that was only present in the traditional analyses, the overlap between treatments was highly similar, this was also true for Fungal (ITS) and Eukaryote (18S) communities. Arthropod communities in traditional analyses did show a difference from compositional approaches, where the difference between bladed and harvested sites was more pronounced. Permanova results were consistent between approaches for all metabarcode targets. Betadispersion in traditional approaches was a bit more consistent between treatments than compositional approaches, particularly in bacterial and fungal (ITS) communities, where there were no significant differences for any treatments.  


```{r CODA-RDA-1, include=FALSE}
clr.rda.plots <- list()

clr.rda.plotting <- data.frame()

plotcount <- 1

for(ds_name in names(coda.clrs.res[grep("_1", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        #if(grepl("Genus|functional", ds_name)){next}
        
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##RDA of the clr values
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ]# %>%
          # mutate(harvest = factor(as.character(harvest), levels = c("Unharvested", "Stem Only", "Full Tree", "Stumped", "Bladed")))
        
        rda.res <- rda(d.clr.abund, char_data[, c("harvest")])
        ## unscaled, is ok b/c all in same units
        
        clr.rda.plotting <- rbind(clr.rda.plotting, data.frame(data.frame(t(summary(rda.res)$cont$importance)) %>%
                                                                 rownames_to_column("PC") %>% 
                                                                 mutate(const = ifelse(grepl("PC", PC), "unconstrained", "constrained")) %>%
                                                                 group_by(const) %>% 
                                                                 summarize(proportion.explained = sum(Proportion.Explained)), model = ds_name))
        
        rda.vecs <- rda.res$CCA$biplot %>% as.data.frame()
        rownames(rda.vecs) <- gsub("^Y", "", rownames(rda.vecs))
        
        ## Prepare plotting variables
        xlab.val <- paste("RDA1", round(summary(rda.res)$cont$importance[2, 1], 3))
        ylab.val <- paste("RDA2", round(summary(rda.res)$cont$importance[2, 2], 3))
        
        ## extract the same way base biplot for princomp works
        plottingdat <- as.data.frame(rda.res$CCA$wa) %>%
          dplyr::select(RDA1, RDA2) %>%
          rownames_to_column("sample") %>% 
          left_join(char_data, by="sample") %>%
          dplyr::select(-sample)
          
        ## Find Hulls of data
      hulls = Ordination_hulls(plottingdat, c("harvest"))
      
      if(grepl("Genus|functional", ds_name)){next}
      
       clr.rda.plots[[plotcount]] <-ggplot(plottingdat, aes(RDA1, RDA2, group=harvest))+
   geom_point(data=plottingdat, aes(color=harvest, shape = harvest), alpha=0.5, size = 2) + 
   geom_polygon(data=hulls, aes(group=hullgrp, fill = harvest, alpha = harvest, color=harvest)) +
   geom_text(inherit.aes = FALSE, data = rda.vecs, aes(label=row.names(rda.vecs), x=RDA1, y=RDA2), color="black") +
   geom_segment(inherit.aes = FALSE, data = rda.vecs, aes(x=0, y=0, xend=RDA1, yend=RDA2), color="black",arrow = arrow()) +
   scale_color_manual(values= cust.cols[c(2, 3, 1, 6, 8)], breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   scale_fill_manual(values= cust.cols[c(2, 3, 1, 6, 8)], breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   scale_shape_manual(values= c(2, 3, 1, 5, 6), breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   scale_alpha_manual(values= c(0.2, 0.2, 0.2, 0.2, 0.2), breaks = levels(plottingdat$harvest)[c(2,3,1,4,5)])+
   xlab(xlab.val)+
   ylab(ylab.val)+
   theme_minimal()+
     ggtitle(ds_name) 
   
names(clr.rda.plots)[plotcount] <- ds_name
        plotcount <- plotcount +1
}


   
rm(plotcount, ds_name, hulls, xlab.val, ylab.val,  rda.res, plottingdat, char_data, temp)
```





### Compositional RDA plots {.tabset}

#### Explained Variance in RDA's{.unnumbered}

```{r RDAsummaryPlots, include=TRUE, fig.cap = "Proportion of community structural variance explained by RDA axes constrained by the harvesting treatments. Model names are described as the targeted gene (16S, 18S, ITS, or, F230), summarization level (ESV, genus, or functional groupings) and the soil layers represented in the rda model (MIN = 0-10 cm depth mineral soil samples, LFH = L, FH, or LFH samples)."}
### create plot that summarizes constrained vs unconstrained variance for each dataset
clr.rda.plotting$const <- factor(as.character(clr.rda.plotting$const), levels = c("unconstrained", "constrained"))

ggplot(clr.rda.plotting[!clr.rda.plotting$const == "unconstrained", ], aes(proportion.explained, model, fill = const)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c(cust.cols[c(2)])) + 
  theme_minimal() + 
  theme(legend.position = "none")+
  xlab("Proportion of variance explained") + 
  ylab("")

rm(clr.rda.plotting)
```

#### Bacterial (16S){.unnumbered}

```{r RDA1}
clr.rda.plots[[1]]
```

#### Fungal (ITS){.unnumbered}

```{r RDA2}
clr.rda.plots[[2]]
```

#### Arthropods (F230){.unnumbered}

```{r RDA3}
clr.rda.plots[[3]]
```

#### Eukaryotes (18S){.unnumbered}

```{r RDA4}
clr.rda.plots[[4]]

rm(clr.rda.plots)
```


## {-}

Slightly less treatment effect was explained in RDA axes when using traditional approaches compared to compositional approaches. Aside from that, the separation of treatments were consistent between approaches.



## Topographic differences


```{r CODA-PCA-3}
clr.pca.plots <- list()

plotcount <- 1

for(ds_name in names(coda.clrs.res[grep("_3", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        #if(grepl("Genus|functional", ds_name)){next}
        
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##PCA of the clr values
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ]# %>%
          # mutate(harvest = factor(as.character(harvest), levels = c("Unharvested", "Stem Only", "Full Tree", "Stumped", "Bladed")))
        
           pca <- prcomp(d.clr.abund)
        
        pca.var.explained <- pca$sdev^2/mvar(d.clr.abund)
        
        ## Prepare plotting variables
        xlab.val <- paste("PCA1", round(pca.var.explained[1], 3))
        ylab.val <- paste("PCA2", round(pca.var.explained[2], 3))
        
        ## extract the same way base biplot for princomp works
        plottingdat <- as.data.frame(pca$x/ncol(pca$x)) %>%
          dplyr::select(PC1, PC2) %>%
          rownames_to_column("sample") %>% 
          left_join(char_data, by="sample") %>%
          dplyr::select(-sample)
        
         ## Find Hulls of data
      hulls = Ordination_hulls(plottingdat, c("topo")) 
      
      if(grepl("Genus|functional", ds_name)){next}
      
       clr.pca.plots[[plotcount]] <-ggplot(plottingdat, aes(PC1, PC2, group=topo))+
   geom_point(data=plottingdat, aes(color=topo, shape = topo), size = 2) + 
   geom_polygon(data=hulls, aes(group=hullgrp, fill = topo, color=topo), alpha = 0.5) +
   scale_color_manual(values= c(cust.cols[8] ,"palegreen4", "burlywood4", "tan"))+
   scale_fill_manual(values= c(cust.cols[8], "palegreen4", "burlywood4", "tan"))+
   scale_shape_manual(values= c(6, 2, 3, 1))+
   xlab(xlab.val)+
   ylab(ylab.val)+
   theme_minimal()+
     ggtitle(ds_name) 
   
       #print(clr.pca.plots[[plotcount]])
       
names(clr.pca.plots)[plotcount] <- ds_name
        plotcount <- plotcount +1
}


   
rm(plotcount, ds_name, char_data, d.clr.abund, hulls, pca, plottingdat, pca.var.explained, xlab.val, ylab.val)
```




### PCA plots {.tabset}

#### Bacterial (16S){.unnumbered}

```{r PCA9}
clr.pca.plots[[1]]
```

#### Fungal (ITS){.unnumbered}

```{r PCA10}
clr.pca.plots[[2]]
```

#### Arthropods (F230){.unnumbered}

```{r PCA11}
clr.pca.plots[[3]]
```

#### Eukaryotes (18S){.unnumbered}

```{r PCA12}
clr.pca.plots[[4]]
rm(clr.pca.plots)
```





```{r CODA-PERMANOVA-3, include=FALSE}
clr.PERM.res <- list()
clr.adon.res <- data.frame()
clr.betadisper.res <- data.frame()
count <- 1

for(ds_name in names(coda.clrs.res[grep("_3", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        #if(grepl("Genus|functional", ds_name)){next}
            if(grepl("Bac", ds_name)){
        distmetric = "bray" ## Relative abundance
  }else{
         distmetric = "jaccard" ## presence/absence
  } 
      
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##char data
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ]
        
        ##OVERALL PERMANOVA
        adon.res.tmp <- adonis(d.clr.abund~topo, data=char_data,  method = distmetric)$aov.tab %>%
           rownames_to_column("Parameter") %>%
           mutate(contrast = "overall",
                  dataset = ds_name) %>% data.frame()
        
        ##OVERALL BETADISPER
        betadisper.res.tmp <- anova(betadisper(vegdist(d.clr.abund,  method = distmetric), char_data$topo)) %>%
           rownames_to_column("Parameter") %>%
           mutate(contrast = "overall",
                  dataset = ds_name) %>% data.frame()
        
        ##PAIRWISE PERMANOVA
        adon.res.tmp <- ortho_adonis(d.clr.abund, topo.contrast.matrix, c("topo"), char_data, method="euclidean") %>%
           mutate(dataset = ds_name, Parameter = "") %>% rbind(adon.res.tmp)
        
        ##PAIRWISE BETADISPER
        
          betadisper.res.tmp <- ortho_betadisper(d.clr.abund, topo.contrast.matrix, c("topo"), char_data, method="euclidean") %>%
           mutate(dataset = ds_name, Parameter = "") %>% rbind(betadisper.res.tmp)
       
          
         clr.adon.res <- rbind(clr.adon.res, adon.res.tmp)
         clr.betadisper.res <- rbind(clr.betadisper.res, betadisper.res.tmp)
         
         
         clr.PERM.res[[count]] <- list(adonis = adonis(d.clr.abund~topo, data=char_data,  method = distmetric), betadisper = betadisper(vegdist(d.clr.abund,  method = distmetric), char_data$topo))
         names(clr.PERM.res)[count] <- ds_name
         count = count + 1
}

num_p_tests <- num_p_tests + length(unique(clr.adon.res$dataset)) + length(unique(clr.adon.res$dataset))
   
rm(count, ds_name, adon.res.tmp, betadisper.res.tmp, char_data)
```

#### PERMANOVA results{.unnumbered}

```{r CODA-adonis-plot-3, fig.cap="Results of orthogonal PERMANOVA tests comparing community structures of the different microtopography features and Bladed sites,orthogonal contrasts are outlined in Table \\@ref(tab:topo-ortho). The variance explained by each comparison (R^2^) is shown on the x axis, and the probability of the effect is along the y axis (Pr (>F)). A black line is used to display p = 0.05."}
adon.plotting <- clr.adon.res[!clr.adon.res$Parameter %in% c("Residuals", "Total", "topo") , ] %>%
  mutate(contrast = factor(as.character(contrast), levels = c("T.B", "T.PF", "P.F")), 
         dataset = gsub("_[[:digit:]]", "", dataset))


ggplot(adon.plotting, aes(R2, Pr..F., color=contrast, shape = contrast)) +
   geom_point(size = 4) + 
   geom_text(aes(label = contrast, y=Pr..F.+.005), show.legend = FALSE) +
   geom_hline(yintercept = 0.05, color="black") + 
   scale_color_manual(values= c(cust.cols[8] ,"tan", "burlywood4"))+
   scale_fill_manual(values= cust.cols[c(cust.cols[8] ,"tan", "burlywood4")])+
   scale_shape_manual(values= c(2, 1, 3, 4, 5, 6))+
   theme_minimal() +
   facet_wrap(~dataset, scales = "free") + 
   xlab(expression("Effect Size (R"^2*")")) + ylab("Pr (>F)")
   
```

#### Betadispersion{.unnumbered}

```{r CODA-bd-plot-3, fig.height=6, fig.cap="Results of orthogonal betadispersion tests comparing community structures of the different microtopography features on full-tree sites, tests are outlined in Table 2. The betadispersion of each treatment (R^2^) is shown on the y axis, and the treatment is shown along the x-axis. Significant differences are indicated in red text (* = p < 0.05, ** = p < 0.01, *** = p < 0.001). A. Bacterial 16S, B. Fungal ITS, C. Arthropod F230, D. Eukaryotic 18S."}

plot.betadisper <- data.frame()

for(ds_name in names(clr.PERM.res)){
   temp <- clr.PERM.res[[ds_name]]$betadisper$distances
   
   ## pull distances for plotting
   temp <- data.frame(temp, sample = names(temp)) %>%
     left_join(metadat, by = "sample") %>%
     mutate(dataset=ds_name) %>%
     rename(distance = temp) 
   
   plot.betadisper <- rbind(plot.betadisper, temp)
     
     
}

## Get min distances for each set
mins  <- plot.betadisper %>%
  group_by(dataset) %>%
  summarize(distance = min(distance))

 ### Pull test result
     temp.sig <- clr.betadisper.res %>%
      filter(contrast != "overall") %>%
     dplyr::select(-Parameter) %>%
     mutate(topo = ifelse(contrast == "T.B", 
                          "Bladed", 
                          ifelse(contrast == "T.PF",
                                 "Trench", 
                                 "Pile"))) %>% 
       group_by(dataset) %>%
     mutate(sig.level = ifelse(Pr..F. < 0.001, "***", ifelse(Pr..F. < 0.01, "**", ifelse(Pr..F. < 0.05, "*", ""))), 
            sig.level = paste(sig.level, ifelse(sig.level !="", contrast, ""))) %>%
       left_join(mins, by = "dataset") %>%
       filter(!grepl("^ $|^$", sig.level)) %>%
           group_by(dataset, topo) %>%
           mutate(distance = unique(distance), 
                  sig.level = str_c(sig.level, collapse = "\n"))

## plot as boxplots with significance
     betadisp.plots <- list()
     count =1
     for(ds in unique(plot.betadisper$dataset)){
           betadisp.plots[[count]] <- ggplot(plot.betadisper[plot.betadisper$dataset == ds, ], aes(topo, distance, fill = topo))+
     geom_boxplot(alpha = 0.5)+
       geom_text(data= temp.sig[temp.sig$dataset == ds,], inherit.aes = F, aes(topo, distance, label = sig.level), color = "red")+
       theme_minimal()+
       scale_fill_manual(values= c(cust.cols[8], "palegreen4", "burlywood4", "tan")) +
                 coord_cartesian(ylim = c(round(min(plot.betadisper[plot.betadisper$dataset == ds, ]$distance)*.8, -1), round(max(plot.betadisper[plot.betadisper$dataset == ds, ]$distance), -1)))
           names(betadisp.plots)[count] <- ds
           count = count +1
     }
     
     ggarrange(betadisp.plots[["Bac16S_3"]], betadisp.plots[["ITS_3"]], betadisp.plots[["F230_3"]], betadisp.plots[["Euk18S_3"]], ncol=2, nrow=2, common.legend = TRUE, labels = "AUTO")
     
```






## {-}

Bacterial (16S) communities had Flat sites that appeared to be outliers in traditional approaches, but not in compositional approaches.  Otherwise, communities separated very similarly to compositional approaches. Bacterial community compositions using the traditional Bray-Curtis relative abundance approach did not have significant differences between mixed and unmixed soils or trench-bladed communities. All remaining tests were significant in traditional and compositional approaches. There were more betadispersion differences in the traditional approaches compared to the compositional approach.


```{r CODA-RDA-3, include=FALSE}
clr.rda.plots <- list()

clr.rda.plotting <- data.frame()

plotcount <- 1

for(ds_name in names(coda.clrs.res[grep("_3", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        #if(grepl("Genus|functional", ds_name)){next}
        
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##RDA of the clr values
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ] %>%
           mutate(topo = factor(topo))
        
        rda.res <- rda(d.clr.abund, char_data[, c("topo")])
        ## unscaled, is ok b/c all in same units
        
        clr.rda.plotting <- rbind(clr.rda.plotting, data.frame(data.frame(t(summary(rda.res)$cont$importance)) %>%
                                                                 rownames_to_column("PC") %>% 
                                                                 mutate(const = ifelse(grepl("PC", PC), "unconstrained", "constrained")) %>%
                                                                 group_by(const) %>% 
                                                                 summarize(proportion.explained = sum(Proportion.Explained)), model = ds_name))
        
        rda.vecs <- rda.res$CCA$biplot %>% as.data.frame()
        rownames(rda.vecs) <- gsub("^Y", "", rownames(rda.vecs))
        
        ## Prepare plotting variables
        xlab.val <- paste("RDA1", round(summary(rda.res)$cont$importance[2, 1], 3))
        ylab.val <- paste("RDA2", round(summary(rda.res)$cont$importance[2, 2], 3))
        
        ## extract the same way base biplot for princomp works
        plottingdat <- as.data.frame(rda.res$CCA$wa) %>%
          dplyr::select(RDA1, RDA2) %>%
          rownames_to_column("sample") %>% 
          left_join(char_data, by="sample") %>%
          dplyr::select(-sample)
          
        ## Find Hulls of data
      hulls = Ordination_hulls(plottingdat, c("topo"))
      
      if(grepl("Genus|functional", ds_name)){next}
      
       clr.rda.plots[[plotcount]] <-ggplot(plottingdat, aes(RDA1, RDA2, group=topo))+
   geom_point(data=plottingdat, aes(color=topo, shape = topo), size = 2) + 
   geom_polygon(data=hulls, aes(group=hullgrp, fill = topo, color=topo), alpha = 0.5) +
   geom_text(inherit.aes = FALSE, data = rda.vecs, aes(label=row.names(rda.vecs), x=RDA1, y=RDA2), color="black") +
   geom_segment(inherit.aes = FALSE, data = rda.vecs, aes(x=0, y=0, xend=RDA1, yend=RDA2), color="black",arrow = arrow()) +
   scale_color_manual(values= c(cust.cols[8], "palegreen4", "burlywood4", "tan"))+
   scale_fill_manual(values= c(cust.cols[8], "palegreen4", "burlywood4", "tan"))+
   scale_shape_manual(values= c(6, 2, 3, 1))+
   xlab(xlab.val)+
   ylab(ylab.val)+
   theme_minimal()+
     ggtitle(ds_name) 
   
names(clr.rda.plots)[plotcount] <- ds_name
        plotcount <- plotcount +1
}


   
rm(plotcount, ds_name, hulls, xlab.val, ylab.val,  rda.res, plottingdat, char_data, temp)
```





### Compositional RDA plots {.tabset}

#### Explained Variance in RDA's{.unnumbered}

```{r RDAsummaryPlots3, include=TRUE, fig.cap= "Proportion of community structural variance explained by RDA axes constrained topographic features. Model names are described as the targeted gene (16S, 18S, ITS, or, F230), summarization level (ESV, genus, or functional groupings) and the soil layers represented in the rda model (MIN = 0-10 cm depth mineral soil samples, LFH = L, FH, or LFH samples)."}
### create plot that summarizes constrained vs unconstrained variance for each dataset
clr.rda.plotting$const <- factor(as.character(clr.rda.plotting$const), levels = c("unconstrained", "constrained"))

ggplot(clr.rda.plotting[!clr.rda.plotting$const == "unconstrained", ], aes(proportion.explained, model, fill = const)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c(cust.cols[c(2)])) + 
  theme_minimal() + 
  theme(legend.position = "none")+
  xlab("Proportion of variance explained") + 
  ylab("")
rm(clr.rda.plotting)
```

#### Bacterial (16S){.unnumbered}

```{r RDA9}
clr.rda.plots[[1]]
```

#### Fungal (ITS){.unnumbered}

```{r RDA10}
clr.rda.plots[[2]]
```

#### Arthropods (F230){.unnumbered}

```{r RDA11}
clr.rda.plots[[3]]
```

#### Eukaryotes (18S){.unnumbered}

```{r RDA12}
clr.rda.plots[[4]]
rm(clr.rda.plots)
```


## {-}

As with permanova and PCA, there was little difference in community structure when comparing traditional and non-traditional approaches of analysis in explained community variance using RDA. 



## Seasonal Patterns


### Differences in diversity


```{r CODA-PCA-2, include=FALSE}
clr.pca.plots <- list()

plotcount <- 1

for(ds_name in names(coda.clrs.res[grep("_2", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        #if(grepl("Genus|functional", ds_name)){next}
            if(grepl("Bac", ds_name)){
        distmetric = "bray" ## Relative abundance
  }else{
         distmetric = "jaccard" ## presence/absence
  }
      
      
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##PCA of the clr values
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ]# %>%
          # mutate(harvest = factor(as.character(harvest), levels = c("Unharvested", "Stem Only", "Full Tree", "Stumped", "Bladed")))
        
           pca <- prcomp(d.clr.abund)
        
        pca.var.explained <- pca$sdev^2/mvar(d.clr.abund)
        
        ## Prepare plotting variables
        xlab.val <- paste("PCA1", round(pca.var.explained[1], 3))
        ylab.val <- paste("PCA2", round(pca.var.explained[2], 3))
        
        ## extract the same way base biplot for princomp works
        plottingdat <- as.data.frame(pca$x/ncol(pca$x)) %>%
          dplyr::select(PC1, PC2) %>%
          rownames_to_column("sample") %>% 
          left_join(char_data, by="sample") %>%
          dplyr::select(-sample)
        
         ## Find Hulls of data
      hulls = Ordination_hulls(plottingdat, c("harvest", "month")) 
      
      if(grepl("Genus|functional", ds_name)){next}
      
       clr.pca.plots[[plotcount]] <-ggplot(plottingdat, aes(PC1, PC2, group=harvest))+ 
   geom_polygon(data=hulls, aes(group=hullgrp, fill = month), alpha = 0.5) +
   geom_point(data=plottingdat, aes(color=harvest, shape = harvest), size = 2) +
   scale_color_manual(values= cust.cols[c(2,1)], breaks = levels(plottingdat$harvest)[c(2,1)])+
   scale_fill_manual(values= c("orchid", "green3", "orange"), breaks = levels(plottingdat$month))+
   scale_shape_manual(values= c(2, 1), breaks = levels(plottingdat$harvest)[c(2,1)])+
   xlab(xlab.val)+
   ylab(ylab.val)+
   theme_minimal()+
     ggtitle(ds_name) 
   
       #print(clr.pca.plots[[plotcount]])
       
names(clr.pca.plots)[plotcount] <- ds_name
        plotcount <- plotcount +1
}


   
rm(plotcount, ds_name, char_data, d.clr.abund, hulls, pca, plottingdat, pca.var.explained, xlab.val, ylab.val)
```

### Compositional PCA plots {.tabset}

#### Bacterial (16S){.unnumbered}

```{r PCA5}
clr.pca.plots[[1]]
```

#### Fungal (ITS){.unnumbered}

```{r PCA6}
clr.pca.plots[[2]]
```

#### Arthropods (F230){.unnumbered}

```{r PCA7}
clr.pca.plots[[3]]
```

#### Eukaryotes (18S){.unnumbered}

```{r PCA8}
clr.pca.plots[[4]]
rm(clr.pca.plots)
```




```{r CODA-PERMANOVA-2, include=FALSE}
clr.PERM.res <- list()
clr.adon.res <- data.frame()
clr.betadisper.res <- data.frame()
count <- 1

for(ds_name in names(coda.clrs.res[grep("_2", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
      #if(grepl("Genus|functional", ds_name)){next}
            if(grepl("Bac", ds_name)){
        distmetric = "bray" ## Relative abundance
  }else{
         distmetric = "jaccard" ## presence/absence
  } 
      
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##char data
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ]%>%
           mutate(harvest = factor(as.character(harvest), levels = c("Unharvested", "Full Tree")))
        
        ##OVERALL PERMANOVA
        adon.res.tmp <- adonis(d.clr.abund~month*harvest, data=char_data,  method = distmetric)$aov.tab %>%
           rownames_to_column("Parameter") %>%
           mutate(test_pair = "overall",
                  dataset = ds_name) %>% data.frame()
        
        ##OVERALL BETADISPER
        betadisper.res.tmp <- anova(betadisper(vegdist(d.clr.abund,  method = distmetric), paste(char_data$harvest, char_data$month))) %>%
           rownames_to_column("Parameter") %>%
           mutate(test_pair = "overall",
                  dataset = ds_name) %>% data.frame()
        
        ##PAIRWISE PERMANOVA
        adon.res.tmp <- pairwise_adonis(d.clr.abund, c("harvest", "month"), char_data,  method = distmetric) %>% 
           mutate(dataset = ds_name, Parameter = "") %>% rbind(adon.res.tmp)
        
        ##PAIRWISE BETADISPER
        
          betadisper.res.tmp <- pairwise_betadisper(d.clr.abund, c("harvest", "month"), char_data,  method = distmetric) %>% 
           mutate(dataset = ds_name, Parameter = "") %>% rbind(betadisper.res.tmp)
       
          
         clr.adon.res <- rbind(clr.adon.res, adon.res.tmp)
         clr.betadisper.res <- rbind(clr.betadisper.res, betadisper.res.tmp)
         
         
         clr.PERM.res[[count]] <- list(adonis = adonis(d.clr.abund~month*harvest, data=char_data,  method = distmetric), betadisper = betadisper(vegdist(d.clr.abund, method = distmetric), paste(char_data$harvest, char_data$month)))
         names(clr.PERM.res)[count] <- ds_name
         count = count + 1
}

num_p_tests <- num_p_tests + length(unique(clr.adon.res$dataset)) + length(unique(clr.adon.res$dataset))
   
rm(count, ds_name, adon.res.tmp, betadisper.res.tmp, char_data)
```


#### PERMANOVA results{.unnumbered}

```{r CODA-adonis-plot-2, fig.cap = "Results of PERMANOVA tests comparing community structures of months (J = June, A = August, O = October) within full tree and unharvested treatments. A) Results of interaction of months and harvest treatments. B) Results of pairwise comparisons. The variance explained by each comparison or model parameter (R^2^) is shown on the x axis, and the probability of the effect is along the y axis (Pr (>F)). A black line is used to represent p = 0.05."}

Perm.plotting <- data.frame()

for(ds in names(clr.PERM.res)){
      Perm.plotting <- rbind(Perm.plotting, clr.PERM.res[[ds]]$adonis$aov.tab %>% rownames_to_column("Parameter") %>% mutate(dataset = ds))
}

plot1 <- ggplot(Perm.plotting %>% filter(!Parameter %in% c("Total", "Residuals")) %>% mutate(dataset = gsub("_2", "", dataset)), aes(R2, `Pr(>F)`, color=Parameter, shape = Parameter)) +
   geom_point(size = 4) + 
   geom_hline(yintercept = 0.05, color="black") + 
   scale_color_manual(name = "", values= cust.cols[c(2, 4, 7)])+
   scale_shape_manual(name = "", values= c(2, 1, 3))+
   theme_minimal() +
   facet_wrap(~dataset, scales = "free") + 
   xlab(expression("Effect Size (R"^2*")")) + ylab("Pr (>F)") +
        theme(legend.position="bottom")

adon.plotting <- clr.adon.res[clr.adon.res$test_pair != "overall", ]

adon.plotting <- adon.plotting %>%
  mutate(month1 = str_extract(test_pair, "^[[:alpha:]]+"), 
         month2 = str_remove(str_extract(test_pair, ":[[:alpha:]]+"), ":"),
         harvest1 = gsub(":|-", "", str_extract(test_pair, "-.+:")), 
         harvest2 = gsub("^.+-", "", str_extract(test_pair, "-.+$"))) %>%
  filter(harvest1 == harvest2) %>%
  mutate(months = factor(ifelse(month2 == "June",
                         paste(str_sub(month2, 1, 1), str_sub(month1, 1, 1), sep = "-"), 
                         ifelse(month1 == "June", 
                                paste(str_sub(month1, 1, 1),str_sub(month2, 1, 1), sep = "-"), 
                                ifelse(month2 == "August", 
                                       paste(str_sub(month2, 1, 1), str_sub(month1, 1, 1), sep = "-"),
                                       paste(str_sub(month1, 1, 1), str_sub(month2, 1, 1), sep = "-")))), 
                         levels= c("J-A", "A-O", "J-O"))) %>%
  rename(harvest = harvest1) %>%
  mutate(harvest = factor(harvest, levels = c("Unharvested", "Full Tree"))) %>%
  dplyr::select(-harvest2, -month1, -month2, -Parameter)

adon.plotting$dataset <- gsub("_.$", "", adon.plotting$dataset)

plot2 <- ggplot(adon.plotting, aes(R2, Pr..F., color=harvest, shape = harvest)) +
   geom_point(size = 4) + 
   geom_text(aes(label = months, y=Pr..F.+.05), show.legend = FALSE) +
   geom_hline(yintercept = 0.05, color="black") + 
   scale_color_manual(name = "", values= cust.cols[c(2, 1)], breaks = levels(adon.plotting$harvest))+
   scale_fill_manual(name = "", values= cust.cols[c(2, 1)], breaks = levels(adon.plotting$harvest))+
   scale_shape_manual(name = "", values= c(2, 1), breaks = levels(adon.plotting$harvest))+
   theme_minimal() +
   facet_wrap(~dataset, scales = "free") + 
   xlab(expression("Effect Size (R"^2*")")) + ylab("Pr (>F)")+
        theme(legend.position="bottom")

ggarrange(plot1, plot2, labels = c("A", "B"))
   
```


#### Betadispersion{.unnumbered}


```{r CODA-bd-plot-2, fig.cap = "Results of pairwise betadispersion tests comparing community structures of months within full tree and unharvested treatments. The betadispersion of each treatment (R^2^) is shown on the y axis, and the treatment is shown along the x-axis. Significant contrasts are indicated at the top of the graph with colour to indicate whether they are harvested or unharvested and significance levels supplied symbolically (* = p < 0.05, ** = p < 0.01, *** = p < 0.001)."}

plot.betadisper <- data.frame()

for(ds_name in names(clr.PERM.res)){
   temp <- clr.PERM.res[[ds_name]]$betadisper$distances
   
   ## pull distances for plotting
   temp <- data.frame(temp, sample = names(temp)) %>%
     left_join(metadat, by = "sample") %>%
     mutate(dataset=ds_name) %>%
     rename(distance = temp) %>%
     mutate(harvest = factor(as.character(harvest), levels = c("Unharvested", "Stem Only", "Full Tree", "Stumped", "Bladed")))
   
   plot.betadisper <- rbind(plot.betadisper, temp)
     
     
}

## Get min distances for each set
maxs  <- plot.betadisper %>%
  group_by(dataset) %>%
  summarize(distance = max(distance))%>%
  mutate(dataset = str_remove(dataset, "_2"))

 ### Pull test result
     temp.sig <- clr.betadisper.res %>%
      filter(test_pair != "overall") %>%
     mutate(month1 = str_extract(test_pair, "^[[:alpha:]]+"), 
         month2 = str_remove(str_extract(test_pair, ":[[:alpha:]]+"), ":"),
         harvest1 = factor(gsub(":|-", "", str_extract(test_pair, "-.+:")), levels = c("Unharvested", "Full Tree")), 
         harvest2 = gsub("^.+-", "", str_extract(test_pair, "-.+$"))) %>%
  filter(harvest1 == harvest2) %>%
  mutate(months = factor(ifelse(month2 == "June",
                         paste(str_sub(month2, 1, 3), str_sub(month1, 1, 3), sep = "-"), 
                         ifelse(month1 == "June", 
                                paste(str_sub(month1, 1, 3),str_sub(month2, 1, 3), sep = "-"), 
                                ifelse(month2 == "August", 
                                       paste(str_sub(month2, 1, 3), str_sub(month1, 1, 3), sep = "-"),
                                       paste(str_sub(month1, 1, 3), str_sub(month2, 1, 3), sep = "-")))), 
                         levels= c("Jun-Aug", "Aug-Oct", "Jun-Oct"))) %>%
     dplyr::select(-Parameter) %>%
       group_by(dataset) %>%
     mutate(sig.level = ifelse(Pr..F. < 0.001, "***", ifelse(Pr..F. < 0.01, "**", ifelse(Pr..F. < 0.05, "*", "")))) %>%
  mutate(dataset = str_remove(dataset, "_2"))%>%
           group_by(dataset) %>%
           arrange(harvest1, months) %>%
           filter(!sig.level == "") %>%
           mutate(seg1.start = ifelse(month1 == "June", .75, ifelse(month1 == "August", 1.75, 2.75)), 
                  seg1.end = ifelse(month1 == "June", 1.25, ifelse(month1 == "August", 2.25, 3.25)), 
                  seg2.start = ifelse(month2 == "June", .75, ifelse(month2 == "August", 1.75, 2.75)), 
                  seg2.end = ifelse(month2 == "June", 1.25, ifelse(month2 == "August", 2.25, 3.25)), 
                  harvest = harvest1
                  ) %>%
           left_join(maxs, by = "dataset") %>%
           mutate(numcount =n():1, 
                  height = distance *(1+(numcount*.1)))
           

     plot.betadisper <- plot.betadisper %>%
           mutate(month = factor(str_extract(month, ".{3}"), levels = c("Jun", "Aug", "Oct")))
     
## plot as boxplots with significance
     ggplot(plot.betadisper%>%
  mutate(dataset = str_remove(dataset, "_2")), aes(month, distance, fill = harvest))+
     geom_boxplot(alpha = 0.5)+
     geom_text(data= temp.sig, inherit.aes = F, aes(3.5, height, label = sig.level, group= harvest1, color = harvest))+
           geom_segment(data = temp.sig, inherit.aes = F, aes(y = height, yend = height, x = seg1.start, xend = seg1.end, color = harvest)) +
           geom_segment(data = temp.sig, inherit.aes = F, aes(y = height, yend = height, x = seg2.start, xend = seg2.end, color = harvest)) +
       theme_minimal()+
        scale_color_manual(values= cust.cols[c(2, 1)], breaks = levels(adon.plotting$harvest))+
       scale_fill_manual(values= cust.cols[c(2, 1)], breaks = levels(adon.plotting$harvest))+
       scale_shape_manual(values= c(2, 1), breaks = levels(adon.plotting$harvest))+
       facet_wrap(~dataset, scales = "free") + ylab("betadispersion")
```

## {-}

There were similarly weak seasonal effects detected using traditional approaches for all metabarcoding target, but fewer betadispersion differences. In both approaches the betadispersion differences between months, while not necessarily significant, were consistent between harvested and unharvested treatments (e.g., generally, lower betadispersion in june in Eukaryotic 18S and F230 communities). 



```{r CODA-RDA-2, include=FALSE}
clr.rda.plots <- list()

clr.rda.plotting <- data.frame()

plotcount <- 1

for(ds_name in names(coda.clrs.res[grep("_2", names(coda.clrs.res), value = T)])){
        
        ## such poor coverage in genus and functional groups that they are excluded here
        #if(grepl("Genus|functional", ds_name)){next}
            if(grepl("Bac", ds_name)){
        distmetric = "bray" ## Relative abundance
  }else{
         distmetric = "jaccard" ## presence/absence
  }
      
        d.clr.abund <- coda.clrs.res[[ds_name]][[1]]
        
        ##RDA of the clr values
        
        char_data <- metadat[match(rownames(d.clr.abund), metadat$sample), ] %>%
           mutate(harvest = factor(as.character(harvest), levels = c("Unharvested","Full Tree")))
        
        rda.res <- rda(d.clr.abund, char_data[, c("month")], char_data[, c("harvest")])
        ## unscaled, is ok b/c all in same units
        
        clr.rda.plotting <- rbind(clr.rda.plotting, data.frame(data.frame(t(summary(rda.res)$cont$importance)) %>%
                                                                 rownames_to_column("PC") %>% 
                                                                 mutate(const = ifelse(grepl("PC", PC), "unconstrained", "constrained")) %>%
                                                                 group_by(const) %>% 
                                                                 summarize(proportion.explained = sum(Proportion.Explained)), model = ds_name))
        
        rda.vecs <- rda.res$CCA$biplot %>% as.data.frame()
        rownames(rda.vecs) <- gsub("^Y", "", rownames(rda.vecs))
        
        ## Prepare plotting variables
        xlab.val <- paste("RDA1", round(summary(rda.res)$cont$importance[2, 1], 3))
        ylab.val <- paste("RDA2", round(summary(rda.res)$cont$importance[2, 2], 3))
        
        ## extract the same way base biplot for princomp works
        plottingdat <- as.data.frame(rda.res$CCA$wa) %>%
          dplyr::select(RDA1, RDA2) %>%
          rownames_to_column("sample") %>% 
          left_join(char_data, by="sample") %>%
          dplyr::select(-sample)
          
        ## Find Hulls of data
      hulls = Ordination_hulls(plottingdat, c("harvest", "month"))
      
      if(grepl("Genus|functional", ds_name)){next}
      
       clr.rda.plots[[plotcount]] <-ggplot(plottingdat, aes(RDA1, RDA2, group=harvest))+ 
   geom_polygon(data=hulls, aes(group=hullgrp, fill = month), alpha = 0.5) +
   geom_point(data=plottingdat, aes(color=harvest, shape = harvest), alpha=0.5, size = 2) +
   geom_text(inherit.aes = FALSE, data = rda.vecs, aes(label=row.names(rda.vecs), x=RDA1, y=RDA2), color="black") +
   geom_segment(inherit.aes = FALSE, data = rda.vecs, aes(x=0, y=0, xend=RDA1, yend=RDA2), color="black",arrow = arrow()) +
   scale_color_manual(values= cust.cols[c(2,1)], breaks = levels(plottingdat$harvest)[c(1,2)])+
   scale_fill_manual(values= c("orchid", "green3", "orange"), breaks = levels(plottingdat$month))+
   scale_shape_manual(values= c(2, 1), breaks = levels(plottingdat$harvest)[c(1, 2)])+
   xlab(xlab.val)+
   ylab(ylab.val)+
   theme_minimal()+
     ggtitle(ds_name) 
   
names(clr.rda.plots)[plotcount] <- ds_name
        plotcount <- plotcount +1
}


   
rm(plotcount, ds_name, hulls, xlab.val, ylab.val,  rda.res, plottingdat, char_data, temp)
```





### Compositional RDA plots {.tabset}

#### Explained Variance in RDA's{.unnumbered}

```{r RDAsummaryPlots2, include=TRUE, fig.cap = "Proportion of community structure variance explained by seasonal differences. Model names are described as the targeted gene (16S, 18S, ITS, or, F230), summarization level (ESV, genus, or functional groupings) and the soil layers represented in the rda model (MIN = 0-10 cm depth mineral soil samples, LFH = L, FH, or LFH samples)."}
### create plot that summarizes constrained vs unconstrained variance for each dataset
clr.rda.plotting$const <- factor(as.character(clr.rda.plotting$const), levels = c("unconstrained", "constrained"))

ggplot(clr.rda.plotting[!clr.rda.plotting$const == "unconstrained", ], aes(proportion.explained, model, fill = const)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c(cust.cols[c(2)])) + 
  theme_minimal() + 
  theme(legend.position = "none")+
  xlab("Proportion of variance explained") + 
  ylab("")
rm(clr.rda.plotting)
```

#### Bacterial (16S){.unnumbered}

```{r RDA5}
clr.rda.plots[[1]]
```

#### Fungal (ITS){.unnumbered}

```{r RDA6}
clr.rda.plots[[2]]
```

#### Arthropods (F230){.unnumbered}

```{r RDA7}
clr.rda.plots[[3]]
```

#### Eukaryotes (18S){.unnumbered}

```{r RDA8}
clr.rda.plots[[4]]
rm(clr.rda.plots)
```

## {-}

Partial RDA with month as constraining variables, and treatment as control variables showed that seasonal variation analyzed through traditional approaches were highly similar to those of non-traditional approaches in both magnitude of variance explained and direction effect of month (Supplemental Materials 2.3.3).  


